ethers-signers/src/wallet/private_key.rs

```diff
@@ -130,12 +130,29 @@ impl FromStr for Wallet<SigningKey> {
     type Err = WalletError;
 
     fn from_str(src: &str) -> Result<Self, Self::Err> {
+        let src = src.strip_prefix("0x").or_else(|| src.strip_prefix("0X")).unwrap_or(src);
         let src = hex::decode(src)?;
         let sk = SigningKey::from_bytes(&src)?;
         Ok(sk.into())
     }
 }
 
+impl TryFrom<&str> for Wallet<SigningKey> {
+    type Error = WalletError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        value.parse()
+    }
+}
+
+impl TryFrom<String> for Wallet<SigningKey> {
+    type Error = WalletError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        value.parse()
+    }
+}
```

Comment:
there's no transient impl when there's `FromStr`?

I guess we can add these

Reply:
The reason I added this is to enable user to have something like this:

```rust
fn create_client(url: &str, priv_key: impl TryInto<LocalWallet>) -> SignerMiddleware<Provider<Http>, LocalWallet>> {
    let wallet = priv_key.try_into().unwrap();
    // ...
}
```

And be able to pass as argument `&str`, `String`, `SigningKey`, and `SecretKey<Secp256k1>`.